app.ts

```diff
@@ -0,0 +1,233 @@
+let headingColumns: any = document.querySelector("#column-headings-container"); //Headings;
+let infoColumns: any = document.querySelector("#info-columns-container"); //Information;
+let debounce = (func: any, delay: number) => {
+  let timer: number;
+  return function () {
+    clearTimeout(timer);
+
+    timer = setTimeout(() => {
+      func();
+    }, delay);
+  };
+};
+
+let tryCatch = (func: any) => {
+  try {
+    func;
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+function createNavigation() {
+  let recordNav: any = document.querySelector("#record-navigation-container"); //Navigation area;
+  recordNav.innerHTML = `
+  <div class="navigation-btns">
+    <button class="first-page-btn">First Page</button>  
+    <button class="previous-records-btn">Previous</button>
+    <button class="next-records-btn">Next</button>
+    <button class="last-page-btn">Last Page</button>  
+    <button onclick="recordSelection()" id="confirmation-btn">Get Record</button>
+  </div>
+  <div class="current-page-container">
+    <p class=current-page></p>
+  </div>
+  `;
+}
+
+function createHeadingGrid(headings: any) {
+  let headingsData: any = `<h1 class="column-heading">${headings}</h1>`;
+  headingColumns.innerHTML += headingsData;
+}
+
+function headingRowCreation() {
+  fetch("http://localhost:2050/columns", {
+    method: "GET",
+    headers: { "Content-Type": "application/json" },
+  })
+    .then((response) => response.text())
+    .then((data) => {
+      let headingDataList = JSON.parse(data);
+      let headings: string;
+
+      for (let i = 0; i < headingDataList.length; i++) {
+        headings = headingDataList[i];
+        createHeadingGrid(headings);
+      }
+      dataRowCreation(0);
+    });
+  createNavigation();
+}
+
+function dynamicGrid(columnData: any) {
+  // Creates the row that the info will display and adds it to the infoColumnsArea.
+  let infoDataRow = `<div id="info-row-${columnData[0]}" class="info-rows"></div>`;
+  infoColumns.innerHTML += infoDataRow;
+  // Gets the created rows.
+  let finalInfoDataRow: any = document.querySelector("#info-row-" + columnData[0] + ".info-rows");
+
+  // Loops through
+  for (let x = 0; x < columnData.length; x++) {
+    let infoData = `<p class="info-row-data">${columnData[x]}</p>`;
+    finalInfoDataRow.innerHTML += infoData;
+  }
+}
+
+let nextPrevious = (fromNumber: number) => {
+  let numberOfRows = Math.floor(window.innerHeight / 50);
+  let count: number = 0;
```

Comment:
Remove count. No need to multiply. The next function is supposed to be executed every time it passes debouncing.